Source/SwiftLintFramework/Extensions/File+SwiftLint.swift

```diff
@@ -270,8 +270,11 @@ extension File {
 
     fileprivate func numberOfCommentAndWhitespaceOnlyLines(startLine: Int, endLine: Int) -> Int {
         let commentKinds = SyntaxKind.commentKinds
-        return syntaxKindsByLines[startLine...endLine].filter { kinds in
-            kinds.filter { !commentKinds.contains($0) }.isEmpty
+        let linesAndSyntaxKinds = zip(lines[(startLine - 1)...(endLine - 1)], syntaxKindsByLines[startLine...endLine])
+
+        return linesAndSyntaxKinds.filter { line, kinds in
+            let isLineEmpty = line.content.trimmingCharacters(in: .whitespaces).isEmpty
+            return isLineEmpty ? true : !kinds.filter { commentKinds.contains($0) }.isEmpty
```

Comment:
As a general rule, `filter { contains }.isEmpty` can always be restructured to `contains { contains }`

Reply:
I'll try to improve it.

Reply:
Should be fixed now. Thanks!